### PR TITLE
fix: if `bucket_id` is given, do not create log bucket

### LINKED ISSUE #44: Latest commit converting bucket_id variable from string to list incorrectly reversed the bucket creation check logic
### Describe the Bug

As described in the title, the latest change in the `bucket_id` variable that converted it from string to list erroneously introduced reverse logic on the check of s3 bucket creation resulting in creating a new bucket it should not.

The offending line can be seen [here](https://github.com/cloudposse/terraform-aws-ssm-patch-manager/commit/7e1f000c3da21aa9d7d3d8717ce3381ddce3db45?diff=split&w=0#diff-2a1969bab6d0901d72421faa5a86b43cac342cba7f73aab781e410d3216946bfR4)

Specifically, the previous logic was written as follows :

```
create_log_bucket = local.enabled && var.bucket_id == null
```

Which correctly created a bucket only when the var.bucket_id variable was not populated with an existing bucket ID.

After the latest commit and release, the logic is now as follows :
```
create_log_bucket = local.enabled && length(var.bucket_id) > 0
```

This now incorrectly creates a bucket (evaluates the condition as true) when the var.bucket_id list is not empty instead of evaluating to true when the list IS actually empty.

Please amend the logic to avoid inconsistencies with exiting resources when plan/applying.

### Expected Behavior

Only create a new s3 bucket when `var.bucket_id` list is empty

### Steps to Reproduce

Attempt to create the module, passing it an existing s3 bucket id as variable

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by AdamTylerLynch**
## what

Fixed a bug. Properly evaluate if bucket provided, do not create bucket

## references

Closes #44

### CODE REVIEW

**Review by Gowiem — APPROVED** [review]
Nice, simple fix! Thanks @AdamTylerLynch

File: ssm_log_bucket.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 locals {
   account_id        = join("", data.aws_caller_identity.current[*].account_id)
   aws_partition     = join("", data.aws_partition.current[*].partition)
-  create_log_bucket = local.enabled && length(var.bucket_id) > 0
+  create_log_bucket = local.enabled && length(var.bucket_id) == 0
   bucket_id         = local.create_log_bucket ? var.bucket_id[0] : module.this.id
```
Comment (Gowiem) [inline ssm_log_bucket.tf]: Tests are failing as this logic was all sorts of wrong it seems. Here is the test failure: 

```
        	            	FatalError{Underlying: error while running command: exit status 1; ╷
        	            	│ Error: Invalid index
        	            	│ 
        	            	│   on ../../ssm_log_bucket.tf line 5, in locals:
        	            	│    5:   bucket_id         = local.create_log_bucket ? var.bucket_id[0] : module.this.id
        	            	│     ├────────────────
        	            	│     │ var.bucket_id is empty list of string
        	            	│ 
        	            	│ The given key does not identify an element in this collection value: the
        	            	│ collection has no elements.
```

This should fix. 

```suggestion
  bucket_id         = local.create_log_bucket ? module.this.id : var.bucket_id[0]
```

**Review by Gowiem — CHANGES_REQUESTED** [review]
Small fix to downstream logic.

**Review by Gowiem — APPROVED** [review]
:shipit:

**Comment by Gowiem** [comment]
Tests look good, but we're getting an annoying IAM permissioning error that is causing the failure: 

```
TestExamplesComplete 2024-09-22T21:02:09Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: creating EC2 Instance: operation error EC2: RunInstances, https response error StatusCode: 403, RequestID: 32d14e15-3e0f-41af-95b6-b6c7a769bac5, api error UnauthorizedOperation: You are not authorized to perform this operation. User: arn:aws:sts::799847381734:assumed-role/cptest-test-ue2-sandbox-gha-iam-terratest/terratest is not authorized to perform: ec2:RunInstances on resource: arn:aws:ec2:us-east-2:799847381734:instance/* with an explicit deny in a service control policy. Encoded authorization failure message: TLA7lt_Zjgdw7Mc9cUwQYaiGRJk4K3QD3wm_LfacQ3YoLrVbYdcekBHxmPiC0DgEBFQ54JROGLSZKziRPCX5IjYI-6iddSSK_CPMfuvsRmVdzu-qKrObNmlbDRpR-D8mThpzkkC4Io9CTy6Uh2VbZ4MXL8Fb13d34VrKov0TJ3TCqjCwYI2i91TlKeltFDV50h3MzR9dj09ZULV-q3SDDCj0aBfmlfq3kMMwEzpIfl33wCWSDM0-QvIbbqNDDg5p1hZg9JzFzPVT1KlwY4Q8Zv_RZZY8cMKsHoFMM_cJqFl2ntBlkzi5jcGK7Te7aN-bHHANfIGLI_z5bB0ZF8NMQCiBqgjyz0ldtnH9ZavdMw2XOQhZ_C3_4OSIOCwiKwnwoZOTzzw1Tu75wfKHm-JOJN2ggAhDHm5djT4E5wNV-ohml4Ial03Ym_2TSQxxn706Uxmcd5lhG0Heqcopo-KxKhYsZZwOm4XbOrNXl7jpilR3GoSd6fek3L_JgJZCrHJCPU6U8GehTaQSuYxqaMefoni5w58LXc87SqjKeL4_kpTNWNtxrRwf1P7_V1pH7eSut-iZDNGk1dz6RcU_O46h6t2MfIw5M1onrXkcRdytHUNdBwSgTKd-JQoB2o1IZhljVfwuw84ljafGI3tNkgRBu6iSMSH_gRIc1Qyy5bFsGkLofPRnkEhwPUqAxKX9gpOD_rGWFSJCJ2u3as25IGmz8maXguTVYcJjqt1Bs3W3JGCBOJgC294xQSe5AG5MZ0xA0_pFNXZWU3jLQzlFhrV5nh_J-ynlXGJ3SQohWzy1QbRPlMB08S0ODBJvdXsuSwY-TwZecH-ixcdUmmrym8JItLA5DZrX0kJjDkZu0xOf9tk8SXhSDQ2IlzYiBPwQtepLA2flAqSm6uhuk3otP57tFZoISZnKE91Vj9Isg_M4K-7rqe_pW58oqUujJVP52y3YVH9Y9-A-fXKfeF08IoBgFtuSptrOWjsV9XdqRgY9Sc1sHeSOPWFkvkGVOV32tFYZmm_KM_1c8MmCkk45ZmnZRg5SECg69aa5KgqPKnUT7emANo6-srfAOp9gcZrhfJGXapIZ7jU0jZX_PSU9JH1qJ_UTDm7ERuzX-1cCaYnejXB4FNA2AI6Cn9hXKAhgapQkYm5cZ4M4ZFTZ2todb5THOp0x1Q
│ 
│   with module.ec2_instance.aws_instance.default[0],
│   on .terraform/modules/ec2_instance/main.tf line 109, in resource "aws_instance" "default":
│  109: resource "aws_instance" "default" {
│ 
╵}
```

I'll surface this with the folks at CP and we'll see if there is something they need to do to the test infra. Sadly, it'll be on hold until we get those green. Apologies for the wait @AdamTylerLynch!

**Comment by Gowiem** [comment]
@AdamTylerLynch quick update: @goruha from the CP team is on this and said he will provide an update / fix this week. We'll get it merged once tests are unblocked. Thanks for the patience!

**Review by goruha — APPROVED** [review]

**Pull request merged**